modules/container-insights/main.tf

```diff
@@ -5,34 +5,72 @@ locals {
+  namespace      = "amazon-cloudwatch"
+  serviceaccount = "amazon-cloudwatch"
   oidc_url       = var.oidc.url
   oidc_arn       = var.oidc.arn
   policy_arns    = ["arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"]
   tags           = var.tags
 }
 
-resource "helm_release" "containerinsights" {
+resource "helm_release" "metrics" {
   count            = var.enabled ? 1 : 0
-  name             = lookup(var.helm, "name", "container-insights")
-  chart            = lookup(var.helm, "chart", "container-insights")
+  name             = "aws-cloudwatch-metrics"
+  chart            = "aws-cloudwatch-metrics"
   version          = lookup(var.helm, "version", null)
-  repository       = lookup(var.helm, "repository", join("/", [path.module, "charts"]))
-  namespace        = local.namespace
+  repository       = lookup(var.helm, "repository", "https://aws.github.io/eks-charts")
+  namespace        = "amazon-cloudwatch"
   create_namespace = true
   cleanup_on_fail  = lookup(var.helm, "cleanup_on_fail", true)
 
   dynamic "set" {
     for_each = {
-      "cluster.name"                                              = var.cluster_name
-      "cluster.region"                                            = data.aws_region.current.0.name
-      "serviceAccount.name"                                       = local.serviceaccount
-      "serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn" = module.irsa[0].arn[0]
+      "clusterName"                                               = var.cluster_name
+      "serviceAccount.name"                                       = "amazon-cloudwatch"
+      "serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn" = module.irsa-metrics[0].arn[0]
+    }
+    content {
+      name  = set.key
+      value = set.value
+    }
+  }
+}
+
+module "irsa-logs" {
+  source         = "../iam-role-for-serviceaccount"
+  count          = var.enabled ? 1 : 0
+  name           = join("-", ["irsa", "aws-for-fluent-bit"])
+  namespace      = "kube-system"
+  serviceaccount = "aws-for-fluent-bit"
+  oidc_url       = var.oidc.url
+  oidc_arn       = var.oidc.arn
+  policy_arns    = ["arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"]
+  tags           = var.tags
+}
+
+resource "helm_release" "logs" {
+  count           = var.enabled ? 1 : 0
+  name            = "aws-for-fluent-bit"
+  chart           = "aws-for-fluent-bit"
+  version         = lookup(var.helm, "version", null)
+  repository      = lookup(var.helm, "repository", "https://aws.github.io/eks-charts")
+  namespace       = "kube-system"
+  cleanup_on_fail = lookup(var.helm, "cleanup_on_fail", true)
+
+  dynamic "set" {
+    for_each = {
+      "cloudWatch.enabled"                                        = true
+      "cloudWatch.region"                                         = data.aws_region.current.0.name
+      "cloudWatch.logGroupName"                                   = format("/aws/containerinsights/%s/application", var.cluster_name)
```

Comment:
changed log group name to follow the container insights naming rule: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Container-Insights-setup-logs-FluentBit.html

Reply:
If you giveme some time I will try again.

Reply:
I think the issue is fixed now.
thanks

now it creates something like:
```
/aws/containerinsights/CLUSTERNAME/application
```